haskell/cabal.bzl

```diff
@@ -838,6 +848,88 @@ be listed in `srcs` (crucially, including the `.cabal` file).
 """,
 )
 
+haskell_cabal_test = rule(
+    _haskell_cabal_test_impl,
+    test = True,
+    attrs = {
+        "test_name": attr.string(
+            doc = "Cabal test component name. Defaults to the value of the name attribute.",
+        ),
+        "srcs": attr.label_list(
+            allow_files = True,
+            doc = "All files required to build the package, including the Cabal file.",
+        ),
+        "deps": attr.label_list(
+            aspects = [haskell_cc_libraries_aspect],
+            doc = "Package build dependencies. Note, setup dependencies need to be declared separately using `setup_deps`.",
+        ),
+        "setup_deps": attr.label_list(
+            aspects = [haskell_cc_libraries_aspect],
+            doc = "Dependencies for custom setup Setup.hs.",
+        ),
+        "cabalopts": attr.string_list(
+            doc = """Additional flags to pass to `Setup.hs configure`. Subject to make variable expansion.
+
+            Use `--ghc-option=OPT` to configure additional compiler flags.
+            Use `--haddock-option=--optghc=OPT` if these flags are required for haddock generation as well.
+            """,
+        ),
+        "compiler_flags": attr.string_list(
+            doc = """DEPRECATED. Use `cabalopts` with `--ghc-option` instead.
+
+            Flags to pass to Haskell compiler, in addition to those defined the cabal file. Subject to Make variable substitution.""",
+        ),
+        "tools": attr.label_list(
+            cfg = "host",
+            allow_files = True,
+            doc = """Tool dependencies. They are built using the host configuration, since
+            the tools are executed as part of the build.""",
+        ),
+        "flags": attr.string_list(
+            doc = "List of Cabal flags, will be passed to `Setup.hs configure --flags=...`.",
+        ),
+        "_cabal_wrapper": attr.label(
+            executable = True,
+            cfg = "host",
+            default = Label("@rules_haskell//haskell:cabal_wrapper"),
+        ),
+        "_cc_toolchain": attr.label(
+            default = Label("@bazel_tools//tools/cpp:current_cc_toolchain"),
+        ),
+        "verbose": attr.bool(
+            default = True,
+            doc = "Whether to show the output of the build",
+        ),
+    },
+    toolchains = [
+        "@bazel_tools//tools/cpp:toolchain_type",
+        "@rules_haskell//haskell:toolchain",
+        "@rules_sh//sh/posix:toolchain_type",
+    ],
+    fragments = ["cpp"],
+    doc = """\
+Use Cabal to build a test.
+
+### Examples
+
+  ```bzl
+  haskell_cabal_test(
+      name = "happy",
+      srcs = glob(["**"]),
+  )
+  ```
+
+This rule assumes that the .cabal file defines a test suite component
+with the same name as the target.
```

Comment:
I think this phrasing is a left over from before we added the `exe_name` attribute in `haskell_cabal_binary`. It's become redundant with the docstring on `exe_name`/`test_name`. Could you also remove the one on`haskell_cabal_binary`?
```suggestion
```